blockprod/src/detail/block_maker.rs

```diff
@@ -92,20 +94,65 @@ impl BlockMaker {
         Ok(returned_accumulator)
     }
 
-    pub fn make_block(
+    pub async fn make_block(
         &self,
         current_tip_id: Id<Block>,
-        accumulator: &dyn TransactionAccumulator,
+        transactions: Transactions,
     ) -> Result<Block, BlockProductionError> {
         // TODO: this isn't efficient. We have to create the header first, then see if it obeys consensus rules, then construct the full block
         let current_time = self.time_getter.get_time();
-        let block = Block::new(
-            accumulator.transactions().clone(),
+        let mut block = Block::new(
+            transactions.transactions().clone(),
             current_tip_id.into(),
             BlockTimestamp::from_duration_since_epoch(current_time),
             common::chain::block::ConsensusData::None,
             BlockReward::new(vec![]), // TODO: define consensus and rewards through NetworkUpgrades
         )?;
+
+        block = self
+            .chainstate_handle
+            .call({
+                let chain_config = self.chain_config.clone();
```

Comment:
why clone?

Reply:
This one is a clone, because otherwise it will complain that it's being moved into the closure

Reply:
Usually we use `Arc::clone(&chain_config)` to signify that this isn't a deep clone. The only ugly thing about it is that you have to import the `Arc` module if it's not already imported.

Reply:
Ah, ok I see now. Yep, I'll change it to follow the standard